libs/sdk-common/src/lnurl/specs/pay.rs

```diff
@@ -0,0 +1,472 @@
+    );
+
+    match comment {
+        None => Ok(()),
+        Some(msg) => match msg.len() <= condition_max_comment_len as usize {
+            true => Ok(()),
+            false => Err(LnUrlError::generic(
+                "Comment is longer than the maximum allowed comment length",
+            )),
+        },
+    }
+}
+
+pub fn validate_invoice(user_amount_msat: u64, bolt11: &str, network: Network) -> LnUrlResult<()> {
+    let invoice = parse_invoice(bolt11)?;
+    // Valid the invoice network against the config network
+    validate_network(invoice.clone(), network)?;
+
+    match invoice.amount_msat {
+        None => Err(LnUrlError::generic(
+            "Amount is bigger than the maximum allowed",
+        )),
+        Some(invoice_amount_msat) => match invoice_amount_msat == user_amount_msat {
+            true => Ok(()),
+            false => Err(LnUrlError::generic(
+                "Invoice amount is different than the user chosen amount",
+            )),
+        },
+    }
+}
+
+pub mod model {
+    use aes::cipher::{block_padding::Pkcs7, BlockDecryptMut, BlockEncryptMut, KeyIvInit};
+    use anyhow::Result;
+    use rusqlite::types::{FromSql, FromSqlError, ToSqlOutput};
+    use rusqlite::ToSql;
+    use serde::{Deserialize, Serialize};
+    use thiserror::Error;
+
+    use crate::prelude::specs::pay::{Aes256CbcDec, Aes256CbcEnc};
+    use crate::prelude::*;
+    // use crate::Payment;
+
+    /// Represents a LNURL-pay request.
+    #[derive(Clone, Debug, Serialize, Deserialize)]
+    pub struct LnUrlPayRequest {
+        /// The [LnUrlPayRequestData] returned by [crate::input_parser::parse]
+        pub data: LnUrlPayRequestData,
+        /// The amount in millisatoshis for this payment
+        pub amount_msat: u64,
+        /// An optional comment for this payment
+        pub comment: Option<String>,
+        /// The external label or identifier of the [Payment]
+        pub payment_label: Option<String>,
+    }
+
+    pub enum ValidatedCallbackResponse {
+        EndpointSuccess { data: CallbackResponse },
+        EndpointError { data: LnUrlErrorData },
+    }
+
+    /// Contains the result of the entire LNURL-pay interaction, as reported by the LNURL endpoint.
+    ///
+    /// * `EndpointSuccess` indicates the payment is complete. The endpoint may return a `SuccessActionProcessed`,
+    /// in which case, the wallet has to present it to the user as described in
+    /// <https://github.com/lnurl/luds/blob/luds/09.md>
+    ///
+    /// * `EndpointError` indicates a generic issue the LNURL endpoint encountered, including a freetext
+    /// field with the reason.
+    ///
+    /// * `PayError` indicates that an error occurred while trying to pay the invoice from the LNURL endpoint.
+    /// This includes the payment hash of the failed invoice and the failure reason.
+    #[derive(Clone, Debug, Serialize, Deserialize)]
+    #[allow(clippy::large_enum_variant)]
+    pub enum LnUrlPayResult {
```

Comment:
Is this enum in use? I don't see it in use. Also it has a variant for PayError which indicates it should be located at the breez-sdk level.

Reply:
Done, removed: 64b4dc16a2f3c32b93f7079ebdcc0a9d1499070a

In the SDK level, its represented by the `Wrapped..` version.